### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by JesStruk**
# Description

This is the initial push/pull of the module with all the code and the example included.

# Change overview (tick true):

- [x] Added code for the core module
- [x] Added an example

# Version information: 

- Previous Version: `1.0.0`

- Next Version based on [Semantic Versioning](https://semver.org/#summary) (see above): `1.0.0`
- `1.0.0` because it is the initial update of the module

# How Has This Been Tested?

- [x] Tested with Terraform Apply in an dedicated environment for the test

# Checklist:

- [ ] I have run tests and documented them above
- [x] I have performed a self-review of my own code
- [ ] I have updated the documentation

### CODE REVIEW

File: locals.tf
Comment (QBY-ChristianHartmann) [inline locals.tf:1]: use variable with default  "platform" instead

File: main.tf
Comment (QBY-ChristianHartmann) [inline main.tf:1]: increase priority by 10?

File: variables.tf [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  validation {
+    condition = alltrue(
+      [for value in var.ip_address_az_dc : can(regex("^(\\d{1,3}[.]){3}(\\d{1,3}[/]\\d{1,3}){1}$", value))]
+    )
+    error_message = "The provided ip address does not match the syntax of ddd.ddd.ddd.ddd/ddd"
+  }
+}
+
+#TODO:
+variable "ip_address_onprem_dc" {
```
Comment (QBY-ChristianHartmann) [inline variables.tf]: variable names if possible without abbreviation. Write onpremises instead of onprem

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  description = "value"
+}
+
+variable "default_location" {
+  type        = string
+  description = "The default location used for this module."
+  nullable    = false
+}
+
+variable "ip_address_az_dc" {
```
Comment (QBY-ChristianHartmann) [inline variables.tf]: variable names if possible without abbreviation. Write azure instead of az

File: variables.tf [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  default     = null
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "The name of the resource group in which the firewall policy and the azure firewall are located."
+  nullable    = false
+}
+
+variable "responsibility" {
```
Comment (QBY-ChristianHartmann) [inline variables.tf:13]: variable name and description don't match. Not really clear that resposibilty is a part of a name

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  description = "value"
+  validation {
+    condition = alltrue(
+      [for value in var.ip_address_alz : can(regex("^(\\d{1,3}[.]){3}(\\d{1,3}[/]\\d{1,3}){1}$", value))]
+    )
+    error_message = "The provided ip address does not match the syntax of ddd.ddd.ddd.ddd/ddd"
+  }
+}
+
+variable "ip_address_DNSPrivateResolver" {
```
Comment (QBY-ChristianHartmann) [inline variables.tf]: variables lowercase

File: main.tf [hunk 5]
```diff
@@ -1 +1,79 @@
...
+  firewall_policy_id = var.firewall_policy_id
+  priority           = 100
 
+  network_rule_collection {
+    name     = "rc-DomainController-${var.stage}"
+    priority = 100
+    action   = "Allow"
+
+    rule {
+      name                  = "rl-alz-to-dc-inbound"
```
Comment (QBY-ChristianHartmann) [inline main.tf]: similiar to nsg rules: allow_source_to_target_direction

File: main.tf [hunk 6]
```diff
@@ -1 +1,79 @@
+resource "azurerm_firewall_policy_rule_collection_group" "policy_collection" {
```
Comment (QBY-ChristianHartmann) [inline main.tf]: check naming of resources : https://confluence.one4all.lan/pages/viewpage.action?spaceKey=AMS&title=Terraform+directive#Terraformdirective-Resources

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+variable "default_location" {
+  type        = string
+  description = "The default location used for this module."
+  nullable    = false
+}
+
+variable "ip_address_az_dc" {
+  type        = set(string)
+  description = "value"
```
Comment (QBY-ChristianHartmann) [inline variables.tf]: description missing in several variables

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  validation {
+    condition = alltrue(
+      [for value in var.ip_address_onprem_dc : can(regex("^(\\d{1,3}[.]){3}(\\d{1,3}[/]\\d{1,3}){1}$", value))]
+    )
+    error_message = "The provided ip address does not match the syntax of ddd.ddd.ddd.ddd/ddd"
+  }
+}
+
+variable "ip_address_alz" {
```
Comment (QBY-ChristianHartmann) [inline variables.tf]: addess of the alz? :o

**Review by QBY-ChristianHartmann — COMMENTED** [review]
general fine. Just some style things.

File: main.tf
Comment (JesStruk) [reply main.tf:1]: Done!

File: locals.tf
Comment (JesStruk) [reply locals.tf:1]: Done!

File: variables.tf (hunk 3 above)
Comment (JesStruk) [reply variables.tf:13]: Changes the description so the name is clear.

File: main.tf (hunk 6 above)
Comment (JesStruk) [reply main.tf]: Changed to "this"

File: variables.tf (hunk 8 above)
Comment (QBY-ChristianHartmann) [reply variables.tf]: please name application_lz to be clear

File: ip_groups.tf [hunk 9]
```diff
@@ -0,0 +1,47 @@
+resource "azurerm_ip_group" "ipg_azure_dc" {
```
Comment (QBY-ChristianHartmann) [inline ip_groups.tf]: remove the ipg.

File: variables.tf (hunk 8 above)
Comment (JesStruk) [reply variables.tf]: named it applicationlandingzone

File: ip_groups.tf (hunk 9 above)
Comment (JesStruk) [reply ip_groups.tf]: Done!

**Review by QBY-ChristianHartmann — APPROVED** [review]

**Pull request merged**